src/platform/packages/shared/kbn-object-utils/src/unflatten_object.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { set } from '@kbn/safer-lodash-set';
+import { DedotObject } from '@kbn/utility-types';
+
+export function unflattenObject<T extends Record<string, any>>(
```

Comment:
I think that we won't fully unflatten a scenario when we have 2 nested flattened objects:

```js
const myObj = {
  'my.flattened.parent.key': {
    'an.internal.key': 1,
  },
};
```

Reply:
Issue: https://github.com/elastic/kibana/issues/214994